components/formats-common/src/loci/common/URLHandle.java

```diff
@@ -64,8 +64,10 @@
    * Constructs a new URLHandle using the given URL.
    */
   public URLHandle(String url) throws IOException {
-    if (!url.startsWith("http") && !url.startsWith("file:")) {
-      url = "http://" + url;
+    if (!url.startsWith("https") && !url.startsWith("file:")) {
+        url = "https://" + url;
+    }else if (!url.startsWith("http") && !url.startsWith("file:")) {
+        url = "http://" + url;
```

Comment:
My guess was that this code is to guess a valid prefix if one is not already supplied but I may be missing some important intent here.